machine_learning/utils.py

```diff
@@ -0,0 +1,131 @@
+    max_df : float in range [0.0,1.0]
+        To ignore words that have a document frequency than given threshold
+        Parameter represents proportion of documents
+    min_df : int
+        To ignore words that have a document frequency strictly lower than threshold
+        Parameter represents absolute counts of words
+
+    Returns
+    -------
+    4 arrays, 2 feature arrays(train & test) and 2 label arrays(train & test)
+    To know more, check out these resources:
+    http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html
+    http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html
+    http://scikit-learn.org/stable/tutorial/text_analytics/working_with_text_data.html
+    """
+    if (vectorizer_func == 'count'):
+        vectorizer = CountVectorizer(analyzer='word',
+                                     stop_words='english',
+                                     max_features=max_features,
+                                     max_df=max_df,
+                                     min_df=min_df)
+    elif (vectorizer_func == 'tfidf'):
+        vectorizer = TfidfVectorizer(analyzer='word',
+                                     stop_words='english',
+                                     max_features=max_features,
+                                     max_df=max_df,
+                                     min_df=min_df,
+                                     sublinear_tf=True)
+    else:
+        raise ValueError('Please select a valid vectorizer.')
+    #Splitting data into training and testing sets
+    train_text, test_text, train_labels, test_labels = train_test_split(text, labels, test_size=0.1)
+
+    #Transforming text to numerical features
+    train_features = vectorizer.fit_transform(train_text)
+    test_features = vectorizer.transform(test_text)
+
+    return train_features, test_features, train_labels, test_labels
+
+def evaluate(eval_method, y_test, y_predicted):
+    """
+    This function computes evaluation metrics.
+
+    Parameters
+    ----------
+    eval_method : string
+        'accuracy' or 'report', more to be added soon
+    y_test : list
+        List of true labels
+    y_predicted : list
+        List of predicted labels
+
+    Returns
+    -------
+    Eval method = 'accuracy' : float between 0.0 and 1.0
+    Eval method = 'report' : string with detailed classification report
+
+    For more info:
+    http://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html
+    http://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html
+    """
+    if (eval_method == 'accuracy'):
+        return accuracy_score(y_true=y_test, y_pred=y_predicted)
+    elif (eval_method == 'report'):
+        return classification_report(y_true=y_test, y_pred=y_predicted)
+    else :
+        raise ValueError('Please select a valid evalaution method.')
+
+def save_model(clf, filename):
```

Comment:
Ditto here. I have definitely felt the temptation to write conveniences like this, but I think it's better to just direct users to the underlying function. If they already know it, they'll be happy. If they don't know it, you've taught them something that might be useful in other contexts, not just in our library.